### PR TITLE
DLQ notification

### PR DESCRIPTION
**Pull request by Alex-Vaky**

### CODE REVIEW

File: src/app/dead-letter-queue/dead-letter-queue.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    fixture = TestBed.createComponent(DeadLetterQueueComponent);
+    component = fixture.componentInstance;
+  });
+
+  [
+    { call_failed: true, expected_priority: PRIORITY.GREY },
+    { call_failed: false, expected_priority: PRIORITY.RED }
+  ].forEach((inputs) => {
+    it('should create a dead letter queue item appropriately', () => {
```
Comment (Blackbaud-ColbyWhite) [inline src/app/dead-letter-queue/dead-letter-queue.component.spec.ts]: nitpick: might want to include `call_failed` and/or `expected_priority` in the `it` block's description j/ so the test output is distinguishable.

File: src/github/services/dead-letter-queue.service.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            }
+        );
+      });
+    }));
+  }
+
+  private convertToReport(requestInputs: DeadLetterQueueReport,
+                          foundDlqMessages = false,
+                          restCallFailed = false): Promise<DeadLetterQueueReport> {
+    console.log('== convertToReport');
```
Comment (Blackbaud-ColbyWhite) [inline src/github/services/dead-letter-queue.service.ts]: nitpick: do we want to nuke these `console.log`s?

File: src/app/dead-letter-queue/dead-letter-queue.component.spec.ts (hunk 1 above)
Comment (Alex-Vaky) [reply src/app/dead-letter-queue/dead-letter-queue.component.spec.ts]: ah, good call

File: src/github/services/dead-letter-queue.service.ts (hunk 2 above)
Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.service.ts]: yes...

Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.service.ts]: i thought i got them all

Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.service.ts]: fixed

File: src/app/dead-letter-queue/dead-letter-queue.component.spec.ts (hunk 1 above)
Comment (Alex-Vaky) [reply src/app/dead-letter-queue/dead-letter-queue.component.spec.ts]: done

File: src/github/services/dead-letter-queue.services.spec.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      ],
+      providers: [
+        { provide: ConfigService, useValue: fakeConfigSvc },
+        DeadLetterQueueService
+      ]
+    });
+
+    service = TestBed.get(DeadLetterQueueService);
+    spyOn(service, 'getAllConfigurations').and.returnValue({
+      'test-team-name': [{
```
Comment (Alex-Vaky) [inline src/github/services/dead-letter-queue.services.spec.ts]: @Blackbaud-ColbyWhite is there a way for me to use my constant here or no?  kind sucks to maintain the string in two places...

Comment (Blackbaud-ColbyWhite) [reply src/github/services/dead-letter-queue.services.spec.ts]: `spyOn(service, 'getAllConfigurations').and.returnValue({[TEST_TEAM]: blah})` should work.

Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.services.spec.ts]: let me try the `[...]` bit and see if things still go smoothyl

Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.services.spec.ts]: totally worked

File: src/app/action-item/action-item.component.ts [hunk 4]
```diff
@@ -1,19 +1,20 @@
 import { PRIORITY } from '../app.constants';
 import * as moment from 'moment';
+import { ACTION_PRIORITY_IGNORE, ACTION_PRIORITY_NEW, ACTION_PRIORITY_NOW, ACTION_PRIORITY_SOON } from '../../domain/action-item';
 
 export class ActionItemComponent {
   calcPriorityClass(priority: number): String {
     switch (priority) {
-      case 1: {
+      case ACTION_PRIORITY_NOW: {
```
Comment (MatthewMazaika) [inline src/app/action-item/action-item.component.ts]: love this

File: src/domain/action-item.ts [hunk 5]
```diff
@@ -14,19 +20,34 @@ export abstract class BaseActionItem {
   abstract get type();
 }
 
+export class DeadLetterQueue extends BaseActionItem {
+
+  constructor(report: DeadLetterQueueReport) {
+    super();
+    this.created = Date.now();
+    this.priority = report.call_failed ? ACTION_PRIORITY_IGNORE : ACTION_PRIORITY_NOW;
+    this.url = ''; // TODO???
```
Comment (MatthewMazaika) [inline src/domain/action-item.ts]: :eyes:

File: src/github/services/dead-letter-queue.service.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      return request.restCall.then(statusReport => {
+        return this.convertToReport(request.requestData, statusReport.positive_report);
+      }).catch(() => {
+        return this.convertToErrorReport(request.requestData);
+      });
+    }));
+  }
+
+  private get queuesToCheck(): Promise<DeadLetterQueueReport>[] {
+    return [].concat(...this.queuesConfigurations.map((config) => {
```
Comment (MatthewMazaika) [inline src/github/services/dead-letter-queue.service.ts]: i think this'll blow up if my team doesn't have any entries in the DLQ file

File: src/domain/action-item.ts (hunk 5 above)
Comment (Alex-Vaky) [reply src/domain/action-item.ts]: yeah i don't know what to do here.  there isn't really a good programmatic url to provide.  @MatthewMazaika you okay with that just being an empty string?  should I remove the todo?

File: src/github/services/dead-letter-queue.service.ts (hunk 6 above)
Comment (Alex-Vaky) [reply src/github/services/dead-letter-queue.service.ts]: addressed inside `this.queuesConfigurations`

File: src/domain/action-item.ts (hunk 5 above)
Comment (Alex-Vaky) [reply src/domain/action-item.ts]: per our slack convo i expanded the comment with more info

Comment (Blackbaud-ColbyWhite) [reply src/domain/action-item.ts]: this is the URL you go to when you click a DLQ item? i take it each DLQ doesn't have a URL? or maybe the URL isn't easily buildable at this point in the code? 

do we have a help doc and/or README somewhere that explains DLQ and then we can link each item to that? or maybe a doc that explains what a dev is supposed to do when they have something on their DLQ? now that i type all that out, if that kind of doc doesn't exist, maybe it should?

**Comment by MatthewMazaika** [comment]
looks good - love the bits of cleanup as you went through it

looking through some of the changes, we'll likely wanna start to do some better extraction of the ActionItems components - it's looking like each of the sub-type components are pretty much the same when it comes to CSS and HTML - the only things that seem to differ are the text and the icon -  - maybe next :tech-thursday: :sparkling_heart:

File: src/domain/action-item.ts (hunk 5 above)
Comment (Alex-Vaky) [reply src/domain/action-item.ts]: so I could send you to your service in the eng hub today, but you wouldn't be able to do any of the management stuff without a provided supportal url which too me screams configuration smell.

if eng sys had dlq management i oculd easily construct a url.  but there just isn't a good url to put here imo.  ideally you see this action item and immediately go to your dlq in your supportal.

**Pull request merged**